completer.go

```
@@ -1,23 +1,43 @@
 package completer
 
-type Completer map[string]string
+import "fmt"
+
+type Completer struct {
+	aliases   map[string]string
+	originals map[string]struct{}
+}
 
 func NewCompleter() Completer {
-	return Completer(make(map[string]string))
+	return Completer{aliases: make(map[string]string), originals: make(map[string]struct{})}
 }
 
-func (c Completer) Add(s string) {
+func (c Completer) Add(s string) error {
+	a := c.aliases
+	if _, ok := a[s]; ok {
+		if _, ok := c.originals[s]; !ok {
+			return fmt.Errorf("unable to add %q: already present as an alias", s)
+		} else {
+			// Duplicate Add() - ignore
+			return nil
```

Adding a duplicate probably indicates an error elsewhere. I'd return a real error here.